x/rollapp/keeper/util.go

```diff
@@ -0,0 +1,85 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v6/modules/apps/transfer/types"
+	ibctypes "github.com/cosmos/ibc-go/v6/modules/light-clients/07-tendermint/types"
+
+	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+// ExtractRollappIDAndTransferPacketFromData extracts the rollapp ID and fungible token from the packet data
+// Returns an empty string if the rollapp is not found
+func (k Keeper) ExtractRollappIDAndTransferPacketFromData(
+	ctx sdk.Context,
+	data []byte,
+	rollappPortOnHub string,
+	rollappChannelOnHub string,
+) (string, *transfertypes.FungibleTokenPacketData, error) {
+	// no-op if the packet is not a fungible token packet
+	packet := new(transfertypes.FungibleTokenPacketData)
+	if err := types.ModuleCdc.UnmarshalJSON(data, packet); err != nil {
+		return "", packet, errorsmod.Wrapf(err, "unmarshal packet data")
+	}
+
+	rollapp, err := k.ExtractRollappFromChannel(ctx, rollappPortOnHub, rollappChannelOnHub)
+	if err != nil {
+		return "", packet, err
+	}
+
+	if rollapp == nil {
+		return "", packet, nil
+	}
+
+	return rollapp.RollappId, packet, nil
+}
+
+// ExtractRollappFromChannel extracts the rollapp from the IBC port and channel.
```

Comment:
should use https://github.com/dymensionxyz/dymension/blob/4565f34fe8a8d2cd9e34a42bd6e3cf71ecbfecd9/x/rollapp/keeper/authenticate_packet.go#L21
all the other middlewares use the pattern, it's easy to copy